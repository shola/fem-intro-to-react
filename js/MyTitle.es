import React from 'react';
import ReactDOM from 'react-dom';
const div = React.DOM.div;
const h1 = React.DOM.h1;

// props are like parameters that are passed to a react element
const MyTitle = React.createClass({
    render() {
        return div(null, h1({ style: { color: this.props.color } }, this.props.title));
    }
});

export default MyTitle;
