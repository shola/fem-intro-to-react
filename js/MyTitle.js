var React = require('react');
var ReactDOM = require('react-dom');
var div = React.DOM.div;
var h1 = React.DOM.h1;

var MyTitle = React.createClass({
    render() {
        return div(null, h1({ style: { color: this.props.color } }, this.props.title));
    }
});

// var MyTitleFactory = React.createFactory(MyTitle);

module.exports = MyTitle;
