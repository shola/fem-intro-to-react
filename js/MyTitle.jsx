// @flow
//
import React, { Component, PropTypes } from 'react';

type Props = {
    title: string, // required prop and required value
    color?: ?string // optional prop and optional value
};

function MyTitle(props: Props): React.Element<any> {
    const { title, color = 'red' } = props;
    const style = { color: color };

    return (
        <h1 className="title" style={style}>
            {title}
        </h1>
    );
}

export default MyTitle;
