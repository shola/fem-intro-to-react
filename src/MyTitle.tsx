import * as React from 'react';

interface Props {
    title: string,
    color?: string
}

function MyTitle(props: Props) {
    const { title, color = 'red' } = props;
    const style = { color: color };

    return (
        <h1 className="title" style={style}>
            {title}
        </h1>
    );
}

export default MyTitle;
