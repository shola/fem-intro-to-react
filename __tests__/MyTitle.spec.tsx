import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as Renderer from 'react-test-renderer';
import MyTitle from '../src/MyTitle';

describe('MyTitle View', () => {
    it('Check snapshot for unexpected changes', () => {
        const component = Renderer.create(<MyTitle />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
