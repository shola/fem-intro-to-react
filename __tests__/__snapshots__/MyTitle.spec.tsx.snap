// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyTitle View Check snapshot for unexpected changes 1`] = `
<h1
  className="title"
  style={
    Object {
      "color": "red",
    }
  }
/>
`;
